python/fusion_engine_client/parsers/file_index.py

```diff
@@ -121,7 +121,7 @@ class FileIndex(object):
     #   timestamp is its exact time
     _RAW_DTYPE = np.dtype([('int', '<u4'), ('type', '<u2'), ('offset', '<u8')])
 
-    _DTYPE = np.dtype([('time', '<f8'), ('type', '<u2'), ('offset', '<u8')])
+    _DTYPE = np.dtype([('time', '<f8'), ('type', '<u2'), ('offset', '<u8'), ('message_index', '<u8')])
```

Comment:
I'm a little confused why this needs to be written out into the binary. Can't the `message_index` be determined by the offset you're at into the index? So if each entry is 11 bytes, you know the index is just index_file.tell()/11.

Or if it's being read as a numpy array, it's just the array's index, unless I'm missing something?

Reply:
It makes sense to preserve the index in the reader when you filter (pandas does something similar, but it seems odd to write it to the index file)